pkg/madmin/health.go

```diff
@@ -20,167 +20,650 @@ package madmin
+	"github.com/shirou/gopsutil/disk"
+	"github.com/shirou/gopsutil/host"
+	"github.com/shirou/gopsutil/mem"
+	"github.com/shirou/gopsutil/net"
+	"github.com/shirou/gopsutil/process"
 )
 
-// HealthInfo - MinIO cluster's health Info
-type HealthInfo struct {
-	TimeStamp time.Time       `json:"timestamp,omitempty"`
-	Error     string          `json:"error,omitempty"`
-	Perf      PerfInfo        `json:"perf,omitempty"`
-	Minio     MinioHealthInfo `json:"minio,omitempty"`
-	Sys       SysHealthInfo   `json:"sys,omitempty"`
+// HealthInfoVersion is current health info version.
+const HealthInfoVersion = "1"
+
+// CPU contains system's CPU information.
+type CPU struct {
+	VendorID   string   `json:"vendor_id"`
+	Family     string   `json:"family"`
+	Model      string   `json:"model"`
+	Stepping   int32    `json:"stepping"`
+	PhysicalID string   `json:"physical_id"`
+	ModelName  string   `json:"model_name"`
+	Mhz        float64  `json:"mhz"`
+	CacheSize  int32    `json:"cache_size"`
+	Flags      []string `json:"flags"`
+	Microcode  string   `json:"microcode"`
+	Cores      int      `json:"cores"` // computed
 }
 
-// SysHealthInfo - Includes hardware and system information of the MinIO cluster
-type SysHealthInfo struct {
-	CPUInfo    []ServerCPUInfo    `json:"cpus,omitempty"`
-	DiskHwInfo []ServerDiskHwInfo `json:"drives,omitempty"`
-	OsInfo     []ServerOsInfo     `json:"osinfos,omitempty"`
-	MemInfo    []ServerMemInfo    `json:"meminfos,omitempty"`
-	ProcInfo   []ServerProcInfo   `json:"procinfos,omitempty"`
-	Error      string             `json:"error,omitempty"`
+// CPUs contains all CPU information of a node.
+type CPUs struct {
+	Addr  string `json:"addr"`
+	Error string `json:"error,omitempty"`
+
+	CPUs []CPU `json:"cpus,omitempty"`
 }
 
-// ServerProcInfo - Includes host process lvl information
-type ServerProcInfo struct {
-	Addr      string       `json:"addr"`
-	Processes []SysProcess `json:"processes,omitempty"`
-	Error     string       `json:"error,omitempty"`
+// GetCPUs returns system's all CPU information.
+func GetCPUs(ctx context.Context, addr string) CPUs {
+	infos, err := cpu.InfoWithContext(ctx)
+	if err != nil {
+		return CPUs{
+			Addr:  addr,
+			Error: err.Error(),
+		}
+	}
+
+	cpuMap := map[string][]cpu.InfoStat{}
+	for _, info := range infos {
+		infoStats, found := cpuMap[info.PhysicalID]
+		if !found {
+			infoStats = []cpu.InfoStat{}
+		}
+		cpuMap[info.PhysicalID] = append(infoStats, info)
+	}
+
+	cpus := []CPU{}
+	for _, infoStats := range cpuMap {
+		cpus = append(cpus, CPU{
+			VendorID:   infoStats[0].VendorID,
+			Family:     infoStats[0].Family,
+			Model:      infoStats[0].Model,
+			Stepping:   infoStats[0].Stepping,
+			PhysicalID: infoStats[0].PhysicalID,
+			ModelName:  infoStats[0].ModelName,
+			Mhz:        infoStats[0].Mhz,
+			CacheSize:  infoStats[0].CacheSize,
+			Flags:      infoStats[0].Flags,
+			Microcode:  infoStats[0].Microcode,
+			Cores:      len(infoStats),
+		})
+	}
```

Comment:
```suggestion
	cpuMap := map[string]CPU{}
	for _, info := range infos {
		cpu, found := cpuMap[info.PhysicalID]
		if found {
			cpu.Cores += 1
		} else {
			cpu = CPU{
				VendorID:   info.VendorID,
				Family:     info.Family,
				Model:      info.Model,
				Stepping:   info.Stepping,
				PhysicalID: info.PhysicalID,
				ModelName:  info.ModelName,
				Mhz:        info.Mhz,
				CacheSize:  info.CacheSize,
				Flags:      info.Flags,
				Microcode:  info.Microcode,
				Cores:      1,
			}
		}
		cpuMap[info.PhysicalID] = cpu
	}

	cpus := []CPU{}
	for _, cpu := range cpuMap {
		cpus = append(cpus, cpu)
	}
```

A relatively simpler and potentially more efficient approach..

Reply:
closing as the code has moved to `madmin-go`